### PR TITLE
feat: yarn 2

### PR DESCRIPTION
**Pull request by christophehurpeau**
See #5230

- use YARN_ENABLE_SCRIPTS=0 instead of --ignore-scripts
- disable yarn-deduplicate with warning

### CODE REVIEW

File: lib/manager/npm/post-update/yarn.ts [hunk 1]
```diff
@@ -125,37 +137,45 @@ export async function generateLockFile(
...
-        cwd,
-        env,
-      });
-      stdout += dedupeRes.stdout
-        ? /* istanbul ignore next */ dedupeRes.stdout
-        : '';
-      stderr += dedupeRes.stderr
-        ? /* istanbul ignore next */ dedupeRes.stderr
-        : '';
+      if (yarnMajorVersion < 2) {
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts]: move to outer scope, so `postUpdateOptions` re checked after major version.

File: lib/manager/npm/post-update/yarn.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { readFile } from 'fs-extra';
 import { join } from 'upath';
 import { getInstalledPath } from 'get-installed-path';
+import * as semverVersioning from '../../../versioning/semver';
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts]:
```suggestion
import { api as semver } from '../../../versioning/semver';
```

File: lib/manager/npm/post-update/yarn.ts [hunk 3]
```diff
@@ -86,9 +87,20 @@ export async function generateLockFile(
     if (binarySource === BinarySource.Global) {
       cmd = 'yarn';
     }
-    logger.debug(`Using yarn: ${cmd}`);
+
+    const { stdout: yarnVersion } = await exec(`${cmd} --version`);
+
+    logger.debug(`Using yarn: ${cmd} ${yarnVersion}`);
+
+    const yarnMajorVersion = semverVersioning.api.getMajor(yarnVersion);
```
Comment (viceice) [inline lib/manager/npm/post-update/yarn.ts]:
```suggestion
    const yarnMajorVersion = semver.getMajor(yarnVersion);
```

**Review by viceice — CHANGES_REQUESTED** [review]
Needs test fixes too.

**Comment by rarkins** [comment]
@christophehurpeau thanks for your submission. Please complete the CLA so that I can take a look.

**Comment by christophehurpeau** [comment]
Hello, I don't know how to fix or adapt the test. I'm guessing the error came from the version addition, but I can't find how I should mock it ? Can you help pointing in the right direction ?

**Comment by JamieMagee** [comment]
@christophehurpeau I've fixed the tests. You can find the diff [here](https://gist.github.com/JamieMagee/235950f9c07e0a72ce71a2844432fc3f).

The issue was, we were mocking all `exec` calls, but your code uses `exec` to get the yarn version but we didn't provide a mocked value for it.

**Comment by christophehurpeau** [comment]
Thank you @JamieMagee !! I've push your patch and updated the branch :)

**Review by viceice — DISMISSED** [review]

**Review by JamieMagee — DISMISSED** [review]

**Review by JamieMagee — APPROVED** [review]

**Comment by JamieMagee** [comment]
I'll merge this once all checks pass.

**Pull request merged**